### DISCUSSION TITLE
Analysis of Adverse Events (AE)

### DISCUSSION THREAD

**Discussion by xni7** [SafetyGraphics/safetyGraphics#499] [General]
Starting a new thread to discuss safety questions related to Adverse Events

@MelvinSMunsaka 's [slides](https://www.bassconference.org/tutorials/BASS%202018%20Munsaka.pdf) has an excellent summary. 

With the multi-domain support in the upcoming Version 2, we have the opportunity to correlate AE with other domains or covariates such as Conmeds, Time, Dose, PK, etc.  
![image](https://user-images.githubusercontent.com/31833867/110989039-50666c00-833f-11eb-8819-46f08465f056.png)

**Comment by xni7** [comment]
On the first box: "is there a relationship with other AEs", we can use clustering and textmining methods
[This paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4158870/#!po=55.0000) _Patient stratification and identification of adverse event correlations in the space of 1190 drug related adverse events_ shows some interesting clustering and heatmap results. 

Below is a quick dirty attempt to replicate the "AE network plot".
https://xiao-ni.shinyapps.io/aeTextMining/

@MayaGans @elimillera  
here is the code: https://github.com/xni7/aeTextMining/blob/master/app.R 
also refer to `{tidytext}` text mining package 
https://www.tidytextmining.com/ngrams.html

![image](https://user-images.githubusercontent.com/31833867/110990183-fa92c380-8340-11eb-93d1-846b19dcf6eb.png)

**Reply by JimBuchanan** [reply]
This is reminiscent of a variable based hierarchical cluster analysis described by Zhong et al (2016) [https://pubmed.ncbi.nlm.nih.gov/26907453/](url) as shown in this dendogram.
![image](https://user-images.githubusercontent.com/38111134/110996482-dd55f900-8330-11eb-85d0-3694714b3882.png)
And tree-based scan statistics like this...
![image](https://user-images.githubusercontent.com/38111134/110996685-23ab5800-8331-11eb-9b1d-7d54a6e47c89.png)
[https://pubmed.ncbi.nlm.nih.gov/30617498/](url)
Finding a co-occurrence of two more AE terms is certainly helpful, but we should also think about how to apply a time dimension as well as a sequence. For example, events that co-occur in a close temporal space may be more likely to be related than if they occurred far apart in time. Similarly, the order in which the AEs occur can provide useful information. If event A always precedes event B, that suggests a non-random pattern compared to situations where half of the time A occurs first and half the time B occurs first. A consistent sequential order could also speak to a biological mechanism. If we could develop a co-occurrence methodology that includes a time window as well as displays sequence patterns, that could be very valuable.
